File: frontend/src/app/models/medecin.ts
import { User } from './user';

export interface Medecin {
  id?: number;
  etatCivil?: string;
  nationnalite?: string;
  prenom: string;
  adresse?: string;
  genre: string;
  type?: string;
  consultations?: any[];  // Remplacez `any` par le type approprié si nécessaire
  rendezVous?: any[];     // Remplacez `any` par le type approprié si nécessaire
  bilans?: any[];         // Remplacez `any` par le type approprié si nécessaire
  certificats?: any[];    // Remplacez `any` par le type approprié si nécessaire
  fiches?: any[];         // Remplacez `any` par le type approprié si nécessaire
  user: User;
}

File: Backend/src/main/java/com/fatimetou/patient/entities/Medecin.java
package com.fatimetou.patient.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Medecin {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String etatCivil;
    private String nationnalite;
    private String prenom;
    private String adresse;

    @Column()
    private String genre;
    private String type;

    @OneToMany(mappedBy = "medecin", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Consultation> consultations;

    @OneToMany(mappedBy = "medecin", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<RendezVous> rendezVous;

    @OneToMany(mappedBy = "medecin", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Bilan> bilans;

    @OneToMany(mappedBy = "medecin", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Certificat> certificats;

    @OneToMany(mappedBy = "medecin", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Fiche> fiches;

    @JsonIgnore
    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    private User user;

    @Override
    public String toString() {
        return "Medecin{" +
                "id=" + id +
                ", etatCivil='" + etatCivil + '\'' +
                ", nationnalite='" + nationnalite + '\'' +
                ", prenom='" + prenom + '\'' +
                ", adresse='" + adresse + '\'' +
                ", genre='" + genre + '\'' +
                ", type='" + type + '\'' +
                '}';
    }
}
